lib/Parser.js

```diff
@@ -1006,47 +1006,103 @@ class Parser extends Tapable {
 
 	prewalkForStatement(statement) {
 		if (statement.init) {
-			if (statement.init.type === "VariableDeclaration")
+			if (
+				statement.init.type === "VariableDeclaration" &&
+				statement.init.kind === "var"
```

Comment:
While prewalking, If variable is not block scoped, then we can preWalk it and register it in current scope, else don't preWalk the variable declaration.